### PR TITLE
i18n(de): Update `endpoints.mdx` translation

### PR DESCRIPTION
**Pull request by bennykay**
#### What kind of changes does this PR include?

- Translated content

#### Description

Translate `endpoints.mdx` to german

### CODE REVIEW

**Comment by selfire1** [comment]
I just had a brief look through the changes and noticed "Sie" being mentioned in lines 14, 102 and 134 when it should be "du".

I'll have a more detailed read through later but just wanted to let you know :)

**Review by selfire1 — CHANGES_REQUESTED** [review]
Thanks for your work in translating this page!

I noticed two minor typos, the rest are just code variable names and a subjective preference. I hope it didn't come across as _penibel_, really appreciate your work!

File: src/content/docs/de/core-concepts/endpoints.mdx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+const usernames = ["Sarah", "Chris", "Yan", "Elian"]
+
+export const GET: APIRoute = ({ params, request }) => {
+  const id = params.id;
+  return new Response(
+    JSON.stringify({
+      name: usernames[id]
+    })
+  )
+}
```
[the remark this answers is not in the document]
Comment (bennykay) [reply src/content/docs/de/core-concepts/endpoints.mdx:64]: I looked at the old translations, there the variables are also not translated into German. I think we should not translate code (variables).

File: src/content/docs/de/core-concepts/endpoints.mdx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+      status: 200,
+      headers: {
+        "Content-Type": "application/json"
+      }
+    }
+  );
+}
+```
+
+Diese Funktion wird jede Anfrage beantworten, deren Adresse zur dynamischen Route passt. Wenn wir zum Beispiel zu `/helmet.json` navigieren, wird `params.id` den Wert `helmet` haben. Wenn in der Produktdatenbank `helmet` existiert, wird der Endpunkt ein `Response`-Objekt erzeugen, um mit JSON und einem erfolgreichen [HTTP-Statuscode](https://developer.mozilla.org/en-US/docs/Web/API/Response/status) zu antworten. Ansonsten wird ein anderes `Response`-Objekt erzeugt, um mit einem `404`-Fehler zu antworten.
```
[the remark this answers is not in the document]
Comment (bennykay) [reply src/content/docs/de/core-concepts/endpoints.mdx:132]: I looked at the old translations, there the variables are also not translated into German. I think we should not translate code (variables).

File: src/content/docs/de/core-concepts/endpoints.mdx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    })
+  )
+}
+
+export const ALL: APIRoute = ({ request }) => {
+  return new Response(JSON.stringify({
+      message: `This was a ${request.method}!`
+    })
+  )
+}
```
[the remark this answers is not in the document]
Comment (bennykay) [reply src/content/docs/de/core-concepts/endpoints.mdx:179]: I looked at the old translations, there the variables are also not translated into German. I think we should not translate code (variables).

**Comment by bennykay** [comment]
Thanks for your review, I don't think you are too *penibel*, we all want to create the best possible result 🙂

I don't think we should translate variables in code, but anything that comes back as a response or button labels can be translated.

**Comment by selfire1** [comment]
My reference was the style guidelines of the German manual where it states that variables names ideally are translated:

> Wir übersetzen gerne auch Komponenten-, Klassen- und Variablennamen in Code-Beispielen. So signalisieren wir, dass diese Namen frei definierbar sind und keine "magischen Keywords" von Astro darstellen.
> — [Anleitung für deutsche Übersetzungen – Stilrichtlinien](https://github.com/withastro/docs/blob/main/src/i18n/de/README.md#stilrichtlinien-style-guide)

**Review by selfire1 — APPROVED** [review]

**Comment by hippotastic** [comment]
Hello and welcome @bennykay! Many thanks for your translation PR!

I just looked at your PR and saw that our link checker found some broken links in it - namely fragment links to the previous English headings that have become invalid because the headings are now translated. Can you please still fix the links accordingly?

Also, your translation of the endpoints page has invalidated a link on another page (`/src/content/docs/en/reference/integrations-reference.mdx`), because it linked to `/en/core-concepts/endpoints/#static-file-endpoints`, which must now end in `#endpoints-for-static-files` according to your translation. Can you please include the corresponding change of the file `/src/content/docs/en/reference/integrations-reference.mdx` in your PR, so that no broken links are found at the end?

By the way, you can see the broken links either directly on GitHub in the context of your changed files (https://github.com/withastro/docs/pull/4547/files), or if you are working on your translation locally, you can always run `pnpm lint:linkcheck` locally to see all problems before committing.

If you need any help with fixing the links, please let me know!

File: src/content/docs/de/core-concepts/endpoints.mdx [hunk 4]
```diff
@@ -0,0 +1,223 @@
+---
+title: Endpunkte
+description: Erfahre, wie du Endpunkte erstellen kannst, die jegliche Form von Daten bereitstellen.
+i18nReady: true
+---
+import RecipeLinks from "~/components/RecipeLinks.astro";
+
+Mit Astro kannst du eigene projektspezifische Endpunkte erstellen, die jegliche Form von Daten bereitstellen können. Du kannst damit Bilder generieren, ein RSS-Dokument bereitstellen oder sie als API-Route verwenden, um eine komplette API für deine Seite zu erstellen.
```
Comment (hippotastic) [inline src/content/docs/de/core-concepts/endpoints.mdx:8]:
```suggestion
Mit Astro kannst du eigene projektspezifische Endpunkte erstellen, die jegliche Form von Daten bereitstellen können. Du kannst damit Bilder generieren, ein RSS-Dokument bereitstellen oder sie als API-Route verwenden, um eine komplette API für deine Website zu erstellen.
```

Hier geht es im Originaltext um eine API für die ganze Website ("site"), und nicht nur eine einzelne Unterseite davon, daher ist die Übersetzung "Website" korrekt.

File: src/content/docs/de/core-concepts/endpoints.mdx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+      status: 200
+    })
+  }
+  return new Response(null, { status: 400 });
+}
+```
+
+### Umleitungen
+
+Der Endpunkt-Kontext exportiert ein `redirect()`-Hilfsfunktion ähnlich wie `Astro.redirect`:
```
Comment (hippotastic) [inline src/content/docs/de/core-concepts/endpoints.mdx:205]:
```suggestion
Der Endpunkt-Kontext exportiert eine `redirect()`-Hilfsfunktion ähnlich wie `Astro.redirect`:
```

**Review by hippotastic — COMMENTED** [review]
This looks like a great PR! In addition to my previous comment regarding the broken links, I made two small suggestions.

Once these open topics are fixed, this is ready to merge!

**Comment by bennykay** [comment]
I fixed all broken links and typos. Thank you for your help!

**Comment by hippotastic** [comment]
Looks almost perfect now! However, we still need to fix the incoming link here:
https://github.com/withastro/docs/blob/f3e1ff0ffa9f35db170b691d9ba7cce774102717/src/content/docs/de/reference/integrations-reference.mdx#L146

As you can see at the very end of the line, this file that is currently not part of your PR contains the link `/de/core-concepts/endpoints/#static-file-endpoints`, which must be changed to `/de/core-concepts/endpoints/#endpunkte-für-statische-dateien` so that it points to your translated heading.

Could you please make this change and commit the file into the branch of this PR? This should fix the remaining link issue.

**Review by hippotastic — APPROVED** [review]
LGTM! Welcome to Team Docs!

**Pull request merged**